lib/commands/clean.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+var path    = require('path');
+var chalk   = require('chalk');
+var Command = require('../models/command');
+var Promise = require('../ext/promise');
+var rimraf  = require('rimraf');
+
+module.exports = Command.extend({
+  name: 'clean',
+  description: 'Cleans up the project from downloads(npm/bower), flushes caches and removes build results directories(tmp/ and dist/ by default).',
+  works: 'insideProject',
+  aliases: ['c'],
+
+  availableOptions: [
+    { name: 'dry-run',     type: Boolean, default: false,   aliases: ['d'] },
+    { name: 'verbose',     type: Boolean, default: false,   aliases: ['v'] },
+    { name: 'skip-npm',    type: Boolean, default: true,   aliases: ['sn'] },
+    { name: 'skip-bower',  type: Boolean, default: false,   aliases: ['sb'] },
+    { name: 'output-path', type: path,    default: 'dist/', aliases: ['o'] }
+  ],
+
+  run: function(commandOptions) {
+    var npmDirectory = 'node_modules',
+      bowerDirectory,
+      outputDir = commandOptions.outputPath,
+      tmpDir = 'tmp',
+      bowerCacheClean,
+      npmCacheClean,
+      ui = this.ui,
+      dirsToRemove = [ tmpDir, outputDir ];
+
+    if (!commandOptions.skipBower) {
+      bowerCacheClean = new this.tasks.BowerCacheClean({
+        ui:             ui,
+        analytics:      this.analytics,
+        project:        this.project
+      });
+
+      var bowerConfig = this.bowerConfig || require('bower-config');
+      var config = bowerConfig.read();
+      bowerDirectory = config.directory || 'bower_components';
+
+      dirsToRemove.push(bowerDirectory);
+    }
+
+    if (!commandOptions.skipNpm) {
+      npmCacheClean = new this.tasks.NpmCacheClean({
+        ui: ui,
+        analytics: this.analytics,
+        project: this.project
+      });
+
+      dirsToRemove.push(npmDirectory);
+    }
+
+    return Promise.resolve()
+      .then(function() {
+        if (!commandOptions.skipBower) {
+          return bowerCacheClean.run({
+            verbose: commandOptions.verbose
+          });
+        }
+      })
+      .then(function() {
+        if (!commandOptions.skipNpm) {
+          return npmCacheClean.run({
+            verbose: commandOptions.verbose
+          });
+        }
+      })
+      .then(function() {
+        dirsToRemove.forEach(function(dirToRemove) {
+          rimraf.sync(dirToRemove);
```

Comment:
Any reason to not use `rimraf` in command directly? Feel like something wrong with it, but not sure how to deal with it.
Possibly it should be extracted in something like RemoveDirTask?